pytorch_lightning/plugins/training_type/ddp.py

```diff
@@ -156,6 +161,9 @@ def _call_children_scripts(self):
         os.environ["NODE_RANK"] = str(self.cluster_environment.node_rank())
         os.environ["LOCAL_RANK"] = str(self.cluster_environment.local_rank())
 
+        # create a temporary directory used to synchronize processes on deadlock.
+        os.environ["PL_DDP_SYNC_TMPDIR"] = self._sync_dir = tempfile.mkdtemp()
```

Comment:
I might be missing something, but how is this shared if using DDP over multiple machines? WIll the temp directory be networked?

Reply:
It won't work for multiple machine right now, but we won't have a simple way to test it out right now.
I hope we can find a way to resolve this.

Reply:
okay but it will not block any multi-gpu experiments right? just the "deadlock detection" will do false positives, right?

Reply:
I'm wondering how/if torchelastic handles this. My curiosity will not drive me there .. 😄

Reply:
So does this mean that all multi node DDP runs will trigger deadock exceptions? Is there a way to turn the check off?

Reply:
it depends where the temp dir is located. if it is only local to the machine, like under /tmp, then yes I think they will trigger because the number of files saved to the temp dir will be less than the world size. 
if it is on a shared file system, it will work. 

we could deactivate this check entirely by checking num_nodes > 1.